core/flow/tc_warm_up.go

```diff
@@ -41,64 +41,67 @@ func NewWarmUpTrafficShapingCalculator(rule *FlowRule) *WarmUpTrafficShapingCalc
 		maxToken:          maxToken,
 		slope:             slope,
 		threshold:         rule.Count,
-		storedTokens:      new(uint64),
-		lastFilledTime:    new(uint64),
+		storedTokens:      0,
+		lastFilledTime:    0,
 	}
 
 	return warmUpTrafficShapingCalculator
 }
 
-func (d *WarmUpTrafficShapingCalculator) CalculateAllowedTokens(node base.StatNode, acquireCount uint32, flag int32) float64 {
+func (c *WarmUpTrafficShapingCalculator) CalculateAllowedTokens(node base.StatNode, acquireCount uint32, flag int32) float64 {
 	previousQps := node.GetPreviousQPS(base.MetricEventPass)
-	d.syncToken(previousQps)
+	c.syncToken(previousQps)
 
-	restToken := atomic.LoadUint64(d.storedTokens)
-	if restToken >= d.warningToken {
-		aboveToken := restToken - d.warningToken
-		warningQps := math.Nextafter(1.0/(float64(aboveToken)*d.slope+1.0/d.threshold), math.MaxFloat64)
+	restToken := atomic.LoadInt64(&c.storedTokens)
+	if restToken < 0 {
```

Comment:
There is no situation where the restToken is less than 0.

Reply:
It might have;
```
if atomic.CompareAndSwapInt64(&c.storedTokens, oldValue, newValue) {
		if currentValue := atomic.AddUint64(d.storedTokens, uint64(0-passQps)); currentValue < 0 {			if currentValue := atomic.AddInt64(&c.storedTokens, int64(-passQps)); currentValue < 0 {
```
In concurrency, after `currentValue := atomic.AddInt64(&c.storedTokens, int64(-passQps)); ` might cause  storedTokens is less than 0;